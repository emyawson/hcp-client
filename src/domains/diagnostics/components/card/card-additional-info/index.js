export { CardAdditionalInfo } from './card-additional-info.component';
