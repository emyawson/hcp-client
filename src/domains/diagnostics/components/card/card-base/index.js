export { CardBase } from './card-base.component';
export { CardContent, CardSection } from './card-base.style';
