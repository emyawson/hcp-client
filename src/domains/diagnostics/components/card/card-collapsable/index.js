export { CardCollapsable } from './card-collapsable.component';
