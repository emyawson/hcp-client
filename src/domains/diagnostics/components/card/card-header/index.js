export { CardHeader } from './card-header.component';
export { CardHeaderWrapper } from './card-header.style';
