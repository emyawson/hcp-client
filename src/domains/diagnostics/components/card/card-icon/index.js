export { CardIcon } from './card-icon.component';
