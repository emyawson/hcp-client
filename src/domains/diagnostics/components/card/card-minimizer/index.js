export { CardMinimizer } from './card-minimizer.component';
