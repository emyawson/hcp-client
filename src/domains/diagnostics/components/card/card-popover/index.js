export { CardPopover } from './card-popover.component';
