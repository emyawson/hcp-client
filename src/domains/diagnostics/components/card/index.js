export { Card } from './card.component';
export { CardIcon } from './card-icon';
export { CardMinimizer } from './card-minimizer';
export { CardAdditionalInfo } from './card-additional-info';
export { CardCollapsable } from './card-collapsable';
export { CardHeader, CardHeaderWrapper } from './card-header';
export { CardBase, CardContent, CardSection } from './card-base';
