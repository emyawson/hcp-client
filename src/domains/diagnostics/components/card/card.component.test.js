import React from 'react';
import { shallow } from 'enzyme';

import { Card } from './card.component';

describe('Card component', () => {
  const defaultProps = {
    title: 'Card',
    expandable: false,
  };

  it('renders child content', () => {
    const childWrapper = shallow(
      <Card {...defaultProps}>
        <p className="childWrapper__paragraph">Test</p>
        <p className="childWrapper__paragraph">Test 2</p>
        <p className="childWrapper__paragraph">Test 3</p>
      </Card>,
    );
    expect(childWrapper).toMatchSnapshot();
  });
});
