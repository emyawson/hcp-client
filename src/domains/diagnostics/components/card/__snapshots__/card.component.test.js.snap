// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders child content 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    expandable={false}
    title="Card"
  >
    <p
      className="childWrapper__paragraph"
    >
      Test
    </p>
    <p
      className="childWrapper__paragraph"
    >
      Test 2
    </p>
    <p
      className="childWrapper__paragraph"
    >
      Test 3
    </p>
  </Card>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "cardStyles": Array [
        "default",
      ],
      "children": Array [
        <f1
          validate="Card"
        >
          <CardHeader
            cardStyles={
              Array [
                "default",
              ]
            }
            customHeaderComponent={undefined}
            expandable={false}
            link={undefined}
            title="Card"
          />
        </f1>,
        <Content
          cardStyles={
            Array [
              "default",
            ]
          }
        >
          <p
            className="childWrapper__paragraph"
          >
            Test
          </p>
          <p
            className="childWrapper__paragraph"
          >
            Test 2
          </p>
          <p
            className="childWrapper__paragraph"
          >
            Test 3
          </p>
        </Content>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <CardHeader
            cardStyles={
              Array [
                "default",
              ]
            }
            customHeaderComponent={undefined}
            expandable={false}
            link={undefined}
            title="Card"
          />,
          "validate": "Card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "cardStyles": Array [
              "default",
            ],
            "customHeaderComponent": undefined,
            "expandable": false,
            "link": undefined,
            "title": "Card",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "cardStyles": Array [
            "default",
          ],
          "children": Array [
            <p
              className="childWrapper__paragraph"
            >
              Test
            </p>,
            <p
              className="childWrapper__paragraph"
            >
              Test 2
            </p>,
            <p
              className="childWrapper__paragraph"
            >
              Test 3
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
              "className": "childWrapper__paragraph",
            },
            "ref": null,
            "rendered": "Test",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Test 2",
              "className": "childWrapper__paragraph",
            },
            "ref": null,
            "rendered": "Test 2",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Test 3",
              "className": "childWrapper__paragraph",
            },
            "ref": null,
            "rendered": "Test 3",
            "type": "p",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "cardStyles": Array [
          "default",
        ],
        "children": Array [
          <f1
            validate="Card"
          >
            <CardHeader
              cardStyles={
                Array [
                  "default",
                ]
              }
              customHeaderComponent={undefined}
              expandable={false}
              link={undefined}
              title="Card"
            />
          </f1>,
          <Content
            cardStyles={
              Array [
                "default",
              ]
            }
          >
            <p
              className="childWrapper__paragraph"
            >
              Test
            </p>
            <p
              className="childWrapper__paragraph"
            >
              Test 2
            </p>
            <p
              className="childWrapper__paragraph"
            >
              Test 3
            </p>
          </Content>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <CardHeader
              cardStyles={
                Array [
                  "default",
                ]
              }
              customHeaderComponent={undefined}
              expandable={false}
              link={undefined}
              title="Card"
            />,
            "validate": "Card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "cardStyles": Array [
                "default",
              ],
              "customHeaderComponent": undefined,
              "expandable": false,
              "link": undefined,
              "title": "Card",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "cardStyles": Array [
              "default",
            ],
            "children": Array [
              <p
                className="childWrapper__paragraph"
              >
                Test
              </p>,
              <p
                className="childWrapper__paragraph"
              >
                Test 2
              </p>,
              <p
                className="childWrapper__paragraph"
              >
                Test 3
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
                "className": "childWrapper__paragraph",
              },
              "ref": null,
              "rendered": "Test",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Test 2",
                "className": "childWrapper__paragraph",
              },
              "ref": null,
              "rendered": "Test 2",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Test 3",
                "className": "childWrapper__paragraph",
              },
              "ref": null,
              "rendered": "Test 3",
              "type": "p",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
